pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -234,39 +237,97 @@ func (b *BlobStorage) buildContainerURL() (azblob.ContainerURL, error) {
 }
 
 func (b *BlobStorage) newPipeline(hedging bool) (pipeline.Pipeline, error) {
-	credential, err := azblob.NewSharedKeyCredential(b.cfg.AccountName, b.cfg.AccountKey.Value)
+	if b.cfg.UseManagedIdentity == false {
+		credential, err := azblob.NewSharedKeyCredential(b.cfg.AccountName, b.cfg.AccountKey.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		opts := azblob.PipelineOptions{
```

Comment:
This is the same option from line 273. Consider defining it before the if clause and reusing it.